library/src/main/java/com/joom/grip/FileRegistry.kt

```diff
@@ -23,52 +23,62 @@ import com.joom.grip.mirrors.Type
 import com.joom.grip.mirrors.getObjectTypeByInternalName
 import java.io.Closeable
 import java.io.File
+import java.nio.file.Path
 import javax.annotation.concurrent.ThreadSafe
 
 @ThreadSafe
 interface FileRegistry {
-  operator fun contains(file: File): Boolean
+  operator fun contains(path: Path): Boolean
   operator fun contains(type: Type.Object): Boolean
 
+  fun readClass(type: Type.Object): ByteArray
+
+  @Deprecated("Deprecated.", ReplaceWith("classpathPaths()"))
   fun classpath(): Collection<File>
 
-  fun readClass(type: Type.Object): ByteArray
+  @Deprecated("Deprecated.", ReplaceWith("findTypesForPath(path)"))
   fun findTypesForFile(file: File): Collection<Type.Object>
+
+  @Deprecated("Deprecated.", ReplaceWith("findPathForType(type)"))
   fun findFileForType(type: Type.Object): File?
+
+  fun classpathPaths(): Collection<Path>
+  fun findTypesForPath(path: Path): Collection<Type.Object>
+  fun findPathForType(type: Type.Object): Path?
 }
 
 internal class FileRegistryImpl(
-  classpath: Iterable<File>,
+  classpath: Iterable<Path>,
   private val fileSourceFactory: FileSource.Factory
 ) : FileRegistry, Closeable {
-  private val sources = LinkedHashMap<File, FileSource>()
-  private val filesByTypes = HashMap<Type.Object, File>()
-  private val typesByFiles = HashMap<File, MutableCollection<Type.Object>>()
+  private val sources = LinkedHashMap<Path, FileSource>()
+  private val filesByTypes = HashMap<Type.Object, Path>()
+  private val typesByFiles = HashMap<Path, MutableCollection<Type.Object>>()
 
   init {
     classpath.forEach {
-      it.canonicalFile.let { file ->
-        if (file !in sources) {
-          val fileSource = fileSourceFactory.createFileSource(file)
-          sources.put(file, fileSource)
+      it.let { sourcePath ->
```

Comment:
`canonicalFile` should be replaced with `toRealPath()` here and in other places.

Reply:
toRealPath will throw exception if file did not exist, which breaks unit test. I use toAbsolutePath() instead

Reply:
You should `normalize()` it then.